dataset/test/bin_test.cpp

```diff
@@ -314,8 +330,37 @@ TEST_P(BinTest, bin_by_group) {
   std::get<2>(binned.data().constituents<core::bin<DataArray>>())
       .coords()
       .erase(Dim("group"));
-  const auto edges =
-      makeVariable<double>(Dims{Dim("group")}, Shape{3}, Values{-2, 0, 2});
-  // Using bin coord )instead of event coord) for binning.
-  bin(binned, {edges});
+  // Using bin coord (instead of event coord) for binning.
+  // Edges giving same grouping as existing => data matches
+  auto edges = makeVariable<double>(Dims{Dim("group")}, Shape{6},
+                                    Values{-2, -1, 0, 1, 2, 3});
+  EXPECT_EQ(bin(binned, {edges}).data(), binned.data());
+
+  // Fewer bins than groups
+  edges = makeVariable<double>(Dims{Dim("group")}, Shape{3}, Values{-2, 0, 2});
+  EXPECT_NO_THROW(bin(binned, {edges}));
+}
+
+TEST_P(BinTest, rebin_various_edges_1d) {
```

Comment:
Yes, basically, I could not think of a good way to check, and this was mainly intdended to cover many edge cases in the setup of bin overlaps. Since the parts doing the actual binning are unchanged the other tests hopefull cover those lines of code.

Reply:
Maybe just add a comment with this explanation?

Reply:
You could use `EXPECT_NOTHROW`